### PR TITLE
feat(graph): add graphify knowledge graph tooling

### LINKED ISSUE #263: feat(tooling): integrate graphify CLI for committed repo knowledge graph
## Summary

Integrate [graphify](https://github.com/safishamsi/graphify) (MIT) into agentic-workflow as a local developer tool that transforms the repo into an interactive, queryable knowledge graph. The repo commits the generated artifacts (`graph.html`, `graph.json`, `GRAPH_REPORT.md`) so any contributor can browse them without installing graphify; contributors who want to regenerate the graph install graphify globally and run an npm script.

## Why

The codebase and tool-belt are growing — stages, agents, skills, scripts, ADRs, and cross-linked docs are increasingly hard to navigate by `grep` alone. A committed knowledge graph gives:
- **Onboarding speed** — new contributors and adopters see structure at a glance.
- **Maintainer overview** — visualize cross-cutting changes before refactoring.
- **Read-only browsing** — anyone with the repo can open `graph/graph.html` in a browser; no install required.

This extends the existing `npm run *` tool-belt pattern (verify, doctor, check:*, fix:*) with discovery and structural insight.

## Scope

- **New file:** `scripts/graphify-run.ts` — thin Node/tsx wrapper. Cross-platform PATH check before exec; exits 1 with a documented message if `graphify` is missing.
- **package.json:** new `graph` (full deep build) + `graph:update` (incremental) scripts.
- **`graph/`** at repo root (not in `files` array, naturally excluded from npm publish): committed `graph.html`, `graph.json`, `GRAPH_REPORT.md`.
- **`.gitignore`:** exclude `graph/cache/` (per-machine SHA256 tracking).
- **Docs:** `docs/how-to/use-graphify.md` (install, run, browse, troubleshoot) + per-script doc under `docs/scripts/graphify-run/`.
- **Verify:** `npm pack --dry-run` confirms zero `graph/` entries; `check:links` + `check:frontmatter` cover the new doc.

## Locked decisions

Resolved in PRD-GRAPH-001 §Open questions:

- **Q1 — Config encoding:** No separate config file. Graphify CLI flags only, baked into the wrapper script. Wrapper is permitted (not a config file) for cross-platform error handling.
- **Q2 — Output directory:** `graph/` at repo root. Not in `package.json#files` → naturally excluded from the npm package; no `.npmignore` change needed.

Resolved in DESIGN-GRAPH-001 §Key decisions:

- Wrapper script over inline npm-script flags (cross-platform PATH check; consistent with all other repo scripts using `tsx scripts/*.ts`).
- No ADR required — all decisions are reversible.

Open at implementation (OQ-GRAPH-001): exact graphify `--output-dir` and `--exclude` flag names. Wrapper falls back to `cd graph/` invocation if needed; resolution recorded in `implementation-log.md`.

## Constitution / taxonomy impact

None. Adds a developer tool, not a new track or stage. ADR-0026 (v1.0 taxonomy freeze) unaffected.

## Artifacts

Branch: `feat/graphify-integration` (worktree at `.worktrees/graphify-integration/`, cut from `origin/main`).

Specorator artifacts under `specs/graphify-integration/` (Lean depth — Stages 1+2 stubbed):

- `idea.md` — Lean stub
- `research.md` — Lean stub
- `requirements.md` — PRD-GRAPH-001, 8 EARS REQs, 5 NFRs, success metrics, release criteria
- `design.md` — DESIGN-GRAPH-001, collapsed UX+UI+Arch (no UI surface)
- `spec.md` — SPECDOC-GRAPH-001, 7 SPEC items, 10 edge cases, 12 test scenarios
- `tasks.md` — TASKS-GRAPH-001, 11 tasks (5 dev, 4 qa, 1 human, 1 verify); TDD-ordered

Bootstrap commit: `2b7675f` (`docs(graphify): bootstrap Lean workflow Stages 1-5`).

`/spec:analyze` PASS with 5 minor findings — all resolved in the bootstrap commit.

## Acceptance

- [ ] All 11 T-GRAPH-* tasks complete; DoDs ticked.
- [ ] `npm run graph` and `npm run graph:update` execute cleanly on macOS, Linux, Windows.
- [ ] `npm pack --dry-run` confirms zero `graph/` entries.
- [ ] `graph/graph.html`, `graph/graph.json`, `graph/GRAPH_REPORT.md` committed; combined size ≤ 10 MB.
- [ ] `git status` shows nothing under `graph/cache/` after a graphify run.
- [ ] `docs/how-to/use-graphify.md` published; `check:links` + `check:frontmatter` pass.
- [ ] `npm run verify` green; no regressions.
- [ ] Retrospective filed (`/spec:retro`).

## Tasks (TDD-ordered)

- T-GRAPH-001 📐 Create `graph/` + `.gitkeep`
- T-GRAPH-002 📐 `.gitignore` rule for `graph/cache/`
- T-GRAPH-003 🧪 Unit tests for wrapper (TEST-GRAPH-001..006, 012)
- T-GRAPH-004 🔨 Implement `scripts/graphify-run.ts`
- T-GRAPH-005 🔨 Add `graph` + `graph:update` npm scripts
- T-GRAPH-006 📚 Write `docs/how-to/use-graphify.md`
- T-GRAPH-007 📚 Per-script doc `docs/scripts/graphify-run/`
- T-GRAPH-008 🧪 Integration test: `graph/` excluded from npm pack
- T-GRAPH-009 🧪 Verify `check:links` + `check:frontmatter` pass on new doc
- T-GRAPH-010 🚀 First graphify run; commit artifacts; size check (owner: human)
- T-GRAPH-011 🧪 Full `npm run verify`

## Out of scope

- Serving `graph.html` via GitHub Pages (NG1).
- CI graph regeneration on push (NG2).
- Graphify as an npm dependency (NG3).
- Wrapper API for `graphify query` / `explain` (NG4).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by Luis85**
## Summary

- Adds `scripts/graphify-run.ts` plus `npm run graph` and `npm run graph:update` for regenerating the committed graphify output under `graph/`.
- Commits generated graph artifacts: `graph/graph.html`, `graph/graph.json`, and `graph/GRAPH_REPORT.md`.
- Adds graphify how-to docs, generated script API docs, automation registry entries, tests, and completed lifecycle artifacts for issue #263.

## Verification

- `npm ci`
- `npm run typecheck:scripts`
- `npm run test:scripts`
- `npm run graph`
- `npm run graph:update`
- `npm run check:automation-registry`
- `npm run check:links`
- `npm run check:frontmatter`
- `npm run check:script-docs`
- `npm run build:release-archive -- --out .release-staging`
- `npm pack .\.release-staging --dry-run --json` (confirmed zero `graph/` entries)
- `npm run verify`

## Notes

- Resolves #263.
- OQ-GRAPH-001 resolved against graphifyy 0.7.0: the supported terminal form is `GRAPHIFY_OUT=graph graphify update .`; the originally assumed `--output-dir`, `--exclude`, and `--version` flags are not available.

### CODE REVIEW

File: tests/scripts/release-package-contract.test.ts [hunk 1]
```diff
@@ -314,6 +315,18 @@ test("parseReleasePackageArgs accepts `--archive <dir>` and `--archive=<dir>`",
...
 });
 
+test("TEST-GRAPH-008: graph directory is excluded from package files", () => {
+  const packageJson = JSON.parse(
+    fs.readFileSync(path.join(repoRoot, "package.json"), "utf8"),
+  ) as { files?: string[] };
+
+  assert.ok(Array.isArray(packageJson.files));
+  assert.equal(
+    packageJson.files.some((entry) => entry === "graph/" || entry.startsWith("graph/")),
```
[the remark this answers is not in the document]
Comment (Luis85) [reply tests/scripts/release-package-contract.test.ts]: Addressed in `18f9438c59fc259c1a1cbcd712aa9e73f85f852a`.

The package-files guard now normalizes entries and rejects all of these forms: `graph`, `graph/`, and nested `graph/...` paths. I also added explicit assertions for those cases.

Verification:
- `npm run test:scripts`
- `npm run verify`

File: tests/scripts/release-package-contract.test.ts [hunk 2]
```diff
@@ -314,6 +315,22 @@ test("parseReleasePackageArgs accepts `--archive <dir>` and `--archive=<dir>`",
   });
 });
 
+test("TEST-GRAPH-008: graph directory is excluded from package files", () => {
+  const packageJson = JSON.parse(
+    fs.readFileSync(path.join(repoRoot, "package.json"), "utf8"),
+  ) as { files?: string[] };
+  const includesGraphDirectory = (entry: string) => {
+    const normalized = entry.replace(/\\/g, "/").replace(/\/+$/, "");
+    return normalized === "graph" || normalized.startsWith("graph/");
```
[the remark this answers is not in the document]
Comment (Luis85) [reply tests/scripts/release-package-contract.test.ts]: Addressed in `dffe9f11eb593fb9484f8a49637c5b3bf66b4128`.

The package-files guard now strips leading `./` segments and leading `/` after slash normalization, so `./graph`, `/graph`, and nested prefixed graph paths are rejected alongside `graph`, `graph/`, and `graph/...`.

Verification:
- `npm run test:scripts`
- `npm run verify`
